packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -354,6 +356,23 @@ class XcodeProjectInterpreter {
   }
 }
 
+/// Environment variables prefixed by FLUTTER_XCODE_ will be passed as build configurations to xcodebuild.
+/// This allows developers to pass arbitrary build settings in without the tool needing to make a flag
+/// for or be aware of each one. This could be used to set code signing build settings in a CI
+/// environment without requiring settings changes in the Xcode project.
+List<String> environmentVariablesAsXcodeBuildSettings() {
+  const String xcodeBuildSettingPrefix = 'FLUTTER_XCODE_';
+  final Iterable<MapEntry<String, String>> xcodeEnvironmentVariables = platform.environment.entries.where((MapEntry<String, String> mapEntry) {
+    return mapEntry.key.startsWith(xcodeBuildSettingPrefix);
+  });
+  final Iterable<String> xcodeBuildSettingsKeyValue = xcodeEnvironmentVariables.expand<String>((MapEntry<String, String> mapEntry) {
+    // Remove FLUTTER_XCODE_ prefix from the environment variable to get the build setting.
+    final String trimmedBuildSettingKey = mapEntry.key.substring(xcodeBuildSettingPrefix.length);
+    return <String>['$trimmedBuildSettingKey=${mapEntry.value}'];
+  });
+  return xcodeBuildSettingsKeyValue.toList();
```

Comment:
It's perfectly fine to return a list here, but note that all of the contexts you are using this in can accept an Iterable as well.